iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
```

Comment:
The issues #177 states that the posh_runtime should be tested but the tests here are incomplete. The following tests cases are at least missing:

- findService
   * finding offered service
   * not offered service
   * reoffer service
   * mix up of AnyService, AnyInstance, AnyEvent
- offerService
   * call with non invalid serviceDescription like `{AnyService, AnyInstance, AnyEvent}`
   * call with non existing serviceDescription
   * call offer service twice
   * call it successfully
- stopOfferService
   * call with non invalid serviceDescription like `{AnyService, AnyInstance, AnyEvent}`
   * call with non existing serviceDescription
   * call stop offer service twice
   * call it successfully

Could you please create the previous mentioned tests and maybe complete the list of unit tests since I am not sure if I have thought of all the cases.

Reply:
I think the scope of this task was to add missing unit test(existing tests are at https://github.com/eclipse/iceoryx/blob/master/iceoryx_posh/test/moduletests/test_runtime.cpp)

Also i guess tests suggested by you comes under integration tests, which are already present at 
https://github.com/eclipse/iceoryx/blob/master/iceoryx_posh/test/integrationtests/service_discovery/test_roudi_findservice.cpp

Reply:
I am not sure what the scope of the task is(up for discussion I guess), but some of those test cases make sense and are missing in the files provided by the links (reoffering, offer service twice for example). Also, I am not sure whether those tests should be classified unit- or integrationtests. They require just the posh runtime (or multiple runtimes) so could very well be unit tests. 

In addition, I guess the proposed test of calling offerService with a nonexisting service description means to call it with a service description that was not offered yet, in contrast to calling it twice with the same service description (just for clarification).

Reply:
@elfenpiff what do you mean by "non invalid" ? valid ? :stuck_out_tongue:

Reply:
@elfenpiff  created issue for remaining tests https://github.com/eclipse/iceoryx/issues/240